staging/src/k8s.io/legacy-cloud-providers/azure/azure_wrap.go

```diff
@@ -62,18 +65,44 @@ func checkResourceExistsFromError(err *retry.Error) (bool, *retry.Error) {
 /// getVirtualMachine calls 'VirtualMachinesClient.Get' with a timed cache
 /// The service side has throttling control that delays responses if there're multiple requests onto certain vm
 /// resource request in short period.
-func (az *Cloud) getVirtualMachine(nodeName types.NodeName, crt cacheReadType) (vm compute.VirtualMachine, err error) {
+func (az *Cloud) getVirtualMachine(nodeName types.NodeName, crt cacheReadType) (compute.VirtualMachine, error) {
 	vmName := string(nodeName)
-	cachedVM, err := az.vmCache.Get(vmName, crt)
+	getter := func(vmName string) (*compute.VirtualMachine, error) {
+		cachedVM, err := az.vmCache.Get(vmKey, crt)
+		if err != nil {
+			return nil, err
+		}
+
+		virtualMachines := cachedVM.(*sync.Map)
+		if vm, ok := virtualMachines.Load(vmName); ok {
+			result := vm.(*vmEntry)
+			return result.vm, nil
+		}
+
+		return nil, nil
+	}
+
+	vm, err := getter(vmName)
 	if err != nil {
-		return vm, err
+		return compute.VirtualMachine{}, err
 	}
 
-	if cachedVM == nil {
-		return vm, cloudprovider.InstanceNotFound
+	if vm != nil {
+		return *vm, nil
 	}
 
-	return *(cachedVM.(*compute.VirtualMachine)), nil
+	klog.V(3).Infof("Couldn't find VM with name %s, refreshing the cache", vmName)
```

Comment:
V(2)

Reply:
Maybe I asked a stupid question, how to decide the level of logging?